src/components/Dashboard.vue

```diff
@@ -637,6 +637,77 @@ export default Vue.extend({
       return arr && arr.length !== 0 && arr[0] !== '' ? arr : ['N/A'];
     },
 
+    getCourseCodesArray(): readonly CourseTaken[] {
+      const courses: CourseTaken[] = [];
+      this.semesters.forEach(semester => {
+        semester.courses.forEach(course => {
+          courses.push({
+            code: `${course.lastRoster}: ${course.subject} ${course.number}`,
+            subject: course.subject,
+            courseId: course.crseId,
+            number: course.number,
+            credits: course.credits,
+            roster: course.lastRoster,
+          });
+        });
+      });
+      courses.push(...getCourseEquivalentsFromUserExams(this.user));
+      return courses;
+    },
+
+    getRequirementTypeDisplayName(type: string): string {
+      return type.charAt(0).toUpperCase() + type.substring(1);
+    },
+
+    recomputeRequirements(): void {
+      const groups = computeRequirements(
+        this.getCourseCodesArray(),
+        this.toggleableRequirementChoices,
+        this.user.college,
+        this.user.major,
+        this.user.minor
+      );
+      // Turn result into data readable by requirements menu
+      const singleMenuRequirements = groups.map(group => {
+        const singleMenuRequirement: SingleMenuRequirement = {
+          ongoing: [],
```

Comment:
I know the type definition exists, but maybe to make it easier for the first time seeing a SingleMenuRequirement, you could add a comment for what should be added to the ongoing[] list and completed[] list?

Reply:
@hahnbeelee maybe you can add some short explanation for me. If I try to add stuff there it might create large merge conflicts for you.

Reply:
ok yeah i can definitely do that!